src/lib.rs

```diff
@@ -172,6 +172,29 @@ enum Command<'a> {
     UnknownVCommand,
     /// Set the current thread for future commands, such as `ReadRegister`.
     SetCurrentThread(ThreadId),
+    /// Insert a software breakpoint.
+    InsertSoftwareBreakpoint(u64, u64, Option<Vec<Vec<u8>>>, Option<Vec<Vec<u8>>>),
```

Comment:
I'm starting to think we should use structs and not tuple structs when there is more than a single element, especially when the types are the same.  This isn't the first one though.

Reply:
I may leave this for a followup to either:

* Convert all `Command`s to be `Variant(x)` where x is some struct; or
* Convert all `Command`s to be `Variant{ field1: t }` and so forth.

But I agree with your thinking that this should be cleaned up.

Reply:
Just to be clear, I don't think it has to be done for this patch.  There are a few cases that should be cleaned up.  I like the idea where the variant holds a struct, because in some situations I think we'll want to pass the struct directly on to the handlers.

Reply:
I like this as a design pattern in Rust in general: enum variants with data should contain a single field, and if they need more it should be a struct. (I sorta wish there was a cleaner way to express this in Rust other than `struct S; enum E { S(S) }`